sdk/python/tests/integration/materialization/test_snowflake.py

```diff
@@ -84,15 +88,32 @@ def test_snowflake_materialization_consistency_internal():
         snowflake_environment.data_source_creator.teardown()
 
 
+@pytest.mark.parametrize(
+    "feature_dtype, feast_dtype",
+    [
+        ("string", Array(String)),
+        ("bytes", Array(Bytes)),
+        ("int32", Array(Int32)),
+        ("int64", Array(Int64)),
+        ("float", Array(Float64)),
+        ("bool", Array(Bool)),
+        ("datetime", Array(UnixTimestamp)),
+    ],
+)
+@pytest.mark.parametrize("feature_is_empty_list", [False])
+@pytest.mark.parametrize("online_store", [SNOWFLAKE_ONLINE_CONFIG, "sqlite"])
 @pytest.mark.integration
-def test_snowflake_materialization_consistency_external():
+def test_snowflake_materialization_consistency_internal_with_lists(
```

Comment:
This test is similar to the one above but I had to verify the output myself instead of the shared function because I'm using Array types.

It's a lot of individual tests but if needed I can probably run a single test with multiple features to test each data type. It's just a lot more custom code for the setup.